x-pack/plugins/security_solution/server/lib/risk_engine/routes/risk_score_calculation_route.ts

```diff
@@ -19,72 +20,77 @@ import { riskScoreServiceFactory } from '../risk_score_service';
 import { getRiskInputsIndex } from '../get_risk_inputs_index';
 
 export const riskScoreCalculationRoute = (router: SecuritySolutionPluginRouter, logger: Logger) => {
-  router.post(
-    {
+  router.versioned
+    .post({
       path: RISK_SCORE_CALCULATION_URL,
-      validate: { body: buildRouteValidation(riskScoreCalculationRequestSchema) },
+      access: 'public',
       options: {
-        tags: ['access:securitySolution'],
+        tags: ['access:securitySolution', `access:${APP_ID}-entity-analytics`],
       },
-    },
-    async (context, request, response) => {
-      const siemResponse = buildSiemResponse(response);
-      const securityContext = await context.securitySolution;
-      const coreContext = await context.core;
-      const esClient = coreContext.elasticsearch.client.asCurrentUser;
-      const soClient = coreContext.savedObjects.client;
-      const spaceId = securityContext.getSpaceId();
-      const riskEngineDataClient = securityContext.getRiskEngineDataClient();
-
-      const riskScoreService = riskScoreServiceFactory({
-        esClient,
-        logger,
-        riskEngineDataClient,
-        spaceId,
-      });
-
-      const {
-        after_keys: userAfterKeys,
-        data_view_id: dataViewId,
-        debug,
-        page_size: userPageSize,
-        identifier_type: identifierType,
-        filter,
-        range,
-        weights,
-      } = request.body;
+    })
+    .addVersion(
+      {
+        version: '2023-10-31',
```

Comment:
Why do we have here version as `2023-10-31`, but in other cases just 1?

Reply:
The dated version is for the one public API, the calculation route. This was the same methodology used in https://github.com/elastic/kibana/pull/164851.